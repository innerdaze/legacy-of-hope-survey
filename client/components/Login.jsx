import React, { Component } from 'react'
import PropTypes from 'prop-types'
import LoginForm from 'grommet/components/LoginForm'
import Box from 'grommet/components/Box'
import { appShortTitle } from '../config'

export default class Login extends Component {
  constructor(props) {
    super(props)

    this.state = {
      username: '',
      password: ''
    }

    this.handleLogin = this.handleLogin.bind(this)
  }

  handleLogin(data) {
    this.props.login(data.username, data.password)
  }

  render() {
    return (
      <Box
        full
        justify='center'
        align='center'
        >
        <LoginForm
          title={appShortTitle}
          secondaryText='By Orbis'
          usernameType='text'
          onSubmit={this.handleLogin}
          />
      </Box>
    )
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired
}

Login.defaultProps = {
  login: Function.prototype
}
