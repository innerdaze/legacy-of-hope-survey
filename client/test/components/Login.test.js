import React from 'react'
import Login from '../../components/Login'

let wrapper, login

describe('Testing on Login...', () => {
  beforeEach(() => {
    login = jest.fn()
    wrapper = shallow(<Login login={login} />)
  })

  describe('Login layout', () => {
    test('Expect Login to have parent Box comp and child LoginForm comp', () => {
      expect(wrapper.find('Box')).toHaveLength(1)
      expect(wrapper.find('Box').find('LoginForm')).toHaveLength(1)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect login prop func to called if LoginForm onSubmit is triggered', () => {
      wrapper.find('LoginForm').simulate('submit', {
        username: 'Foo',
        password: 'Bar'
      })

      expect(login).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on state', () => {
    test('Expect default username and password state is both empty string', () => {
      expect(wrapper.state('username')).toEqual('')
      expect(wrapper.state('password')).toEqual('')
      expect(wrapper).toMatchSnapshot()
    })
  })
})
