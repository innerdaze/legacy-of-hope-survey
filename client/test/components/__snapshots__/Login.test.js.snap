// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on Login... Login layout Expect Login to have parent Box comp and child LoginForm comp 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  full={true}
  justify="center"
  pad="none"
  responsive={true}
  tag="div"
>
  <LoginForm
    align="center"
    defaultValues={
      Object {
        "rememberMe": false,
        "username": "",
      }
    }
    errors={Array []}
    onSubmit={[Function]}
    secondaryText="By Orbis"
    title="Site Assistant"
    usernameType="text"
  />
</Box>
`;

exports[`Testing on Login... Test on prop Expect login prop func to called if LoginForm onSubmit is triggered 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  full={true}
  justify="center"
  pad="none"
  responsive={true}
  tag="div"
>
  <LoginForm
    align="center"
    defaultValues={
      Object {
        "rememberMe": false,
        "username": "",
      }
    }
    errors={Array []}
    onSubmit={[Function]}
    secondaryText="By Orbis"
    title="Site Assistant"
    usernameType="text"
  />
</Box>
`;

exports[`Testing on Login... Test on state Expect default username and password state is both empty string 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  full={true}
  justify="center"
  pad="none"
  responsive={true}
  tag="div"
>
  <LoginForm
    align="center"
    defaultValues={
      Object {
        "rememberMe": false,
        "username": "",
      }
    }
    errors={Array []}
    onSubmit={[Function]}
    secondaryText="By Orbis"
    title="Site Assistant"
    usernameType="text"
  />
</Box>
`;
